apps/backend/src/modules/users/user.service.ts

```diff
@@ -1,32 +1,56 @@
+import { ApplicationError } from "~/libs/exceptions/exceptions.js";
+import { type EncryptionService } from "~/libs/modules/encryption/encryption.js";
 import { type Service } from "~/libs/types/types.js";
 import { UserEntity } from "~/modules/users/user.entity.js";
 import { type UserRepository } from "~/modules/users/user.repository.js";
 
+import { UserServiceMessage } from "./libs/enums/user-service-message.enum.js";
 import {
 	type UserGetAllResponseDto,
 	type UserSignUpRequestDto,
 	type UserSignUpResponseDto,
 } from "./libs/types/types.js";
 
 class UserService implements Service {
+	private encryptionService: EncryptionService;
 	private userRepository: UserRepository;
 
-	public constructor(userRepository: UserRepository) {
+	public constructor(
+		userRepository: UserRepository,
+		encryptionService: EncryptionService,
+	) {
 		this.userRepository = userRepository;
+		this.encryptionService = encryptionService;
 	}
 
 	public async create(
 		payload: UserSignUpRequestDto,
 	): Promise<UserSignUpResponseDto> {
+		const { email, name, password } = payload;
+		const existingUser = await this.userRepository.find({ email });
+
+		if (existingUser) {
+			throw new ApplicationError({
```

Comment:
Create a custom UserError
```suggestion
			throw new UserError({
```